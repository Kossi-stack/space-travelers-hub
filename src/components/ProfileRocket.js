import PropTypes from 'prop-types';

const ProfileRocket = ({ rockets }) => (
  <li key={rockets.id} className="profile__li">
    {rockets.rocket_name}
  </li>
);

ProfileRocket.propTypes = {
  rockets: PropTypes.shape({
    id: PropTypes.number.isRequired,
    rocket_name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool,
  }).isRequired,
};

export default ProfileRocket;
