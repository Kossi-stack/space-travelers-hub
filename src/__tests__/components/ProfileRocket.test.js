import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ProfileRocket from '../../components/ProfileRocket';

it('renders Profile Rockets Component correctly', () => {
  const rocket = {
    id: 1,
    rocket_name: 'falcon 1',
    description: 'some description',
    reserved: true,
    flickr_images: [
      'https://imgur.com/DaCfMsj.jpg',
      'https://imgur.com/azYafd8.jpg',
    ],
  };
  const tree = renderer
    .create(
      <Router>
        <ProfileRocket key={rocket.id} rockets={rocket} />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
