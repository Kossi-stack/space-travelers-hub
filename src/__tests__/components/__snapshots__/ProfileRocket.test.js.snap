// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile Rockets Component correctly 1`] = `
<li
  className="profile__li"
>
  falcon 1
</li>
`;
